storage/badger/operation/computation_result.go

```diff
@@ -0,0 +1,45 @@
+package operation
+
+import (
+	"github.com/dgraph-io/badger/v2"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// InsertComputationResult addes given instance of ComputationResult into local BadgerDB.
+func InsertComputationResultUploadStatus(computationResultID flow.Identifier,
+	wasUploadCompleted bool) func(*badger.Txn) error {
+	return insert(makePrefix(codeComputationResults, computationResultID), wasUploadCompleted)
+}
+
+// RemoveComputationResult removes an instance of ComputationResult with given ID.
+func RemoveComputationResultUploadStatus(
+	computationResultID flow.Identifier) func(*badger.Txn) error {
+	return remove(makePrefix(codeComputationResults, computationResultID))
+}
+
+// GetComputationResult returns stored ComputationResult instance with given ID.
+func GetComputationResultUploadStatus(computationResultID flow.Identifier,
+	wasUploadCompleted *bool) func(*badger.Txn) error {
+	return retrieve(makePrefix(codeComputationResults, computationResultID), wasUploadCompleted)
+}
+
+// GetAllComputationResultIDs returns all IDs of stored ComputationResult instances.
+func GetAllComputationResultIDs(computationResultIDs *[]flow.Identifier) func(*badger.Txn) error {
```

Comment:
Does it work as expected? it seems to collect all the entries where value is true?

Reply:
It simply returns all stored items, regardless of the actual values stored (true\false). Since it is in storage layer, I'd keep the implementation use case agnostic, leaving filtering to its callers.

Reply:
ddressed. the data retrieval interface now parameterized with a target status to filter